src/frontend/src/scheduler/worker_node_manager.rs

```diff
@@ -82,10 +82,16 @@ impl WorkerNodeManager {
         self.worker_nodes.read().unwrap().len()
     }
 
+    /// If parallel unit ids is empty, the scheduler may fail to schedule any task and stuck at
+    /// schedule next stage. If we do not return error in this case, needs more complex control
+    /// logic above. Report in this function makes the schedule root fail reason more clear.
     pub fn get_workers_by_parallel_unit_ids(
         &self,
         parallel_unit_ids: &[ParallelUnitId],
     ) -> SchedulerResult<Vec<WorkerNode>> {
+        if parallel_unit_ids.is_empty() {
+            return Err(SchedulerError::EmptyWorkerNodes);
+        }
```

Comment:
Is it possible to have empty parallel_unit_ids? 😇

Reply:
We have a unit test called test_query_should_not_hang_with_empty_worker in query.rs 😢

Reply:
I can understand empty workers. But empty parallel_unit_ids means empty vnode mapping, which I think is impossible? 🤔

Reply:
The vnode_mapping is empty in test_query_should_not_hang_with_empty_worker. You can add a breakpoint in L293 stage.rs. Because if and only if parallel_unit is empty, then we will get empty workers.

If vnode_bitmaps should not be empty when workers are empty, I guess there are some bugs 😇.

Reply:
Oh yes, it uses `create_query`, which has a table scan with `vnode_mapping: Some(vec![])`. I think this is impossible in practice 😄